lib/matplotlib/cm.py

```diff
@@ -331,6 +332,35 @@ def unregister_cmap(name):
     return _cmap_registry.pop(name)
 
 
+@functools.lru_cache(None)
+def _auto_norm_from_scale(scale_cls):
```

Comment:
1) Some scales (most prominently, logit) don't have a norm associated; 2) because everything is wrapped under lru_caches, `_auto_norm_from_scale(LogScale)` is going to be exactly the same object as `LogNorm` (multiple calls to `make_norm_from_scale` or `_auto_norm_from_scale` always return the same norm class).  (I added a test for that.)

Reply:
Can you document that here? Because I couldn't parse that from the code, but maybe my ignorance of how the caches get shared across submodules.  

However, I'm still not understanding what happens if the user puts 'logit' in here.  Are you saying a norm will still be created?

Reply:
The documentation is the third bullet point below ("The returned norm class is memoized and reused for later calls."  But feel free to suggest something clearer).  Yes, a norm class will be dynamically created if the user puts in "logit" (and later calls will reuse the same dynamically created norm class).
This will also work with scales provided by third-parties, e.g. mpl-probscale (which is why I did not go for just an explicit and exhaustive mapping of names to norm classes -- this would not work with third-party scales).

Reply:
> this would not work with third-party scale

Fair enough if we are supporting those as norms, though that sounds risky to me...

> The returned norm class is memoized and reused for later calls.

Maybe just a bit more verbose "the returned norm class is the same as for make_norm_from_scale (so norm='log' will return a LogNorm instance); these are are also memoized..."  (though is there a reason to not use "cached"?  "memoized" is pretty jargon, but maybe my ignorance)

Reply:
Re: 3rd-parties: well, that was also one of the points of the PR (and again, think of this as "supporting putting a third-party scale on a colorbar axis").
Re: docs: sure, reworded, sorry for the jargon :)

Reply:
Edit: actually I had to fix the caching, but it should be good now.

Reply:
I had a bit of confusion initially figuring out exactly what is being cached. The cached "thing" is the class definition. It is not the returned Norm object. So, these two calls will have different LogNormalize objects, which is consistent with how it works currently when no norm is given (a new Normalize object is created each time).
```python
im1 = ax.imshow([[0, 1]], norm="log")
im2 = ax.imshow([[0, 1]], norm="log")
assert im1.norm is not im2.norm
```

Reply:
Good point, I further reworded the docs to clarify that.